core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -751,6 +759,28 @@ public StageExecutionDescriptor getStageExecutionDescriptor()
         }
     }
 
+    public static class OperatorFactoryWithTypes
```

Comment:
Is it possible to add types to operator factory?

Reply:
There are 55 implementations of `OperatorFactory`. Adding types there would be much broader change. I will keep it here for now and we can move it out if there is another use-case that would benefit from types (e.g making operators more plug-gable maybe)?